pyvrp/cpp/ProblemData.h

```diff
@@ -300,24 +374,36 @@ class ProblemData
     [[nodiscard]] size_t numVehicles() const;
 
     /**
-     * Constructs a ProblemData object with the given data. Assumes the list of
-     * clients contains the depot, such that each vector is one longer than the
-     * number of clients.
+     * Constructs a ProblemData object with the given data. The given distance
+     * and duration matrices should have the depots on the lower indices (in
+     * order of the depots argument), followed by the clients (again in order).
      *
-     * @param clients      List of clients (including depot at index 0).
+     * @param clients      List of clients.
+     * @param depots       List of depots.
      * @param vehicleTypes List of vehicle types.
      * @param distMat      Distance matrix.
      * @param durMat       Duration matrix.
      */
     ProblemData(std::vector<Client> const &clients,
+                std::vector<Client> const &depots,
                 std::vector<VehicleType> const &vehicleTypes,
                 Matrix<Distance> distMat,
                 Matrix<Duration> durMat);
 };
 
-ProblemData::Client const &ProblemData::client(size_t client) const
+ProblemData::Client const &ProblemData::location(size_t idx) const
+{
+    return idx < depots_.size() ? depots_[idx] : clients_[idx - depots_.size()];
+}
+
+ProblemData::Client const &ProblemData::client(size_t idx) const
+{
+    return clients_[idx];
+}
+
+ProblemData::Client const &ProblemData::depot(size_t idx) const
 {
-    return clients_[client];
+    return depots_[idx];
 }
```

Comment:
On PyVRP 0.6.3, it is possible to use `ProblemData.client(idx)` for `idx > numClients`, which then returns completely random clients. Can we make this raise a runtime error? Same for depots and location.

Reply:
I'll add an assert to the function and move the error raising to the bindings, so we can avoid the check when this gets called from C++ (which happens a lot), but not when a user interacts with it from Python.